biggus/__init__.py

```diff
@@ -839,10 +891,114 @@ def _getitem_full_keys(self, keys):
             sub_array = self.array
         else:
             sub_array = self.array[tuple(array_keys)]
-        return BroadcastArray(sub_array, new_broadcast_dict)
+        return BroadcastArray(sub_array, new_broadcast_dict, leading_shape)
+
+    @classmethod
+    def broadcast_arrays(cls, array1, array2):
+        """
+        Given two arrays, if broadcasting needs to take place on either arrays
```

Comment:
It's OK. If you're tempted to change it then I'd follow the numpy docs: "Broadcast any number of arrays against each other.", which suggests perhaps "Broadcast two arrays against each other." together with a "Returns" section which states that the arrays will be untouched if possible or wrapped as BroadcastArrays.